code/datums/job_configs/job_configs.dm

```diff
@@ -0,0 +1,131 @@
+// This contains the types of configurations that can be set for each job.
+// Just add a new datum (and add the define name to __defines/jobs.dm) and the applicable procs and you should be good to go.
+// Remember, there's a verb in the Server tab called "Generate Job Configuration" that will generate the config file for you.
+// You don't need to waste time copy-pasting values if you add a datum here. Just use that verb and it'll do it for you.
+// Use the verb. Use the verb. It's your life you're wasting otherwise.
+
+/// Lightweight datum simply used to store the applicable config type for each job such that the whole system is a tad bit more flexible.
+/datum/job_config_type
+	var/name = "DEFAULT"
+
+/// Simply gets the value of the config type for a given job. There can be overrides for special instances on subtypes.
+/datum/job_config_type/proc/get_compile_time_value(datum/job/occupation)
+	SHOULD_CALL_PARENT(FALSE)
+	stack_trace("Attempted to get value for the default job config for [occupation.title] (with config tag [occupation.config_tag])! This is not allowed!")
+
+/// Validate the value of the config type for a given job. There can be overrides for special instances on subtypes.
+/// Isn't meant for in-depth logic, just bare-bones sanity checks. Like: is this number a number? Is this string a string? In-depth sanity checks should be done in the set_current_value proc.
+/// Will return TRUE if the value is valid, FALSE if it is not.
+/datum/job_config_type/proc/validate_value(value)
+	SHOULD_CALL_PARENT(FALSE)
+	stack_trace("Attempted to validate value for the default job config! You're doing something wrong!!")
+
+/// This is the proc that we actually invoke to set the config-based values for each job. Is also intended to handle all in-depth logic checks.
+/datum/job_config_type/proc/set_current_value(datum/job/occupation, value)
+	SHOULD_CALL_PARENT(FALSE)
+	CRASH("Attempted to set value for the default job config for [occupation.title] (with config tag [occupation.config_tag])! This is not allowed!")
+
+/// The number of positions a job can have at any given time.
+/datum/job_config_type/default_positions
+	name = JOB_CONFIG_TOTAL_POSITIONS
+
+/datum/job_config_type/default_positions/get_compile_time_value(datum/job/occupation)
+	if(is_assistant_job(occupation)) // yeah i know it's not the "compile time" value but this is just to remain parity with the old system groooan
+		return -1
+	return initial(occupation.total_positions)
+
+/datum/job_config_type/default_positions/validate_value(value)
+	if(isnum(value))
+		return TRUE
+	return FALSE
+
+/datum/job_config_type/default_positions/set_current_value(datum/job/occupation, value)
+	if(!validate_value(value))
+		return FALSE
+	occupation.total_positions = value
+	return TRUE
```

Comment:
I don't like that all instances of this proc look the exact same.
could validate_value be called on the parent? 

is there a better way of doing this?

Reply:
i fixed it in an upcoming commit and it's effectively the same boilerplate but it's consistent enough of a pattern where i'm okay with it